### DISCUSSION TITLE
Can I build a SPA with solid start?

### DISCUSSION THREAD

**Discussion by CallMeHK** [solidjs/solid-start#1398] [Q&A]
There is no example in the docs - I would like to build a spa and deploy it to github pages. There is [one dev.to article](https://dev.to/lexlohr/deploy-a-solid-start-app-on-github-pages-2l2l) on the subject, but the settings given in the article still generate a server.

**Comment by atk** [comment]
I'm the author of the article you linked. While a server is generated, it only is used once to generate the static HTML output. For the result to become an SPA, the service worker is missing, though, and I haven't tested yet if vite plugins for SPAs will work in this settings.

**Comment by ryansolid** [comment]
Generally with the new presets you can set `ssr: false` and then `server.static: true` and this will give you SPA output. There is a server which we use to render the index.html but I've tested this with providers like Netlify, and Vercel. Along with Github pages using the appropriate presets and it seems to work.

**Comment by ugoa** [comment]
Hi @atk @ryansolid, 

Solid.js newbie here, I am trying to build a client-side only SPA with SolidStart. Here is my config:
```ts
import { defineConfig } from '@solidjs/start/config'

export default defineConfig({
    ssr: false,
    server: {
        static: true,
        prerender: {
            crawlLinks: true,
        },
    },
})
```

and here is my output folder:
![ss](https://github.com/solidjs/solid-start/assets/1934726/92611338-7238-493a-9d6c-8cfeecab2932)

My question is, if I want to use nginx to host it, do I need to copy everything under `.output/public` to nginx?
Appreciate if you could share some input, cheers.

**Reply by DamianKaczmarczykK** [reply]
@ugoa Not sure if you need this info after few months (but it may be useful for others) - I've just managed to deploy `solid-start` app to github pages:
1. In my case, [I needed to add empty .nojekyll](https://github.blog/news-insights/bypassing-jekyll-on-github-pages/) file to include `_build` and `_server` files
2. Also setting `baseURL` was mandatory for github pages to correctly resolve file paths

For some normal nginx configuration, I suppose copying `.output/public` would be sufficient (maybe some files are not needed, dunno)

Here is my example project: https://github.com/DamianKaczmarczykK/backtesting-playground/tree/gh-pages - `compiled` files are under `/docs` catalog
